artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -205,17 +206,23 @@ public PagingStoreImpl(final SimpleString address,
 
       this.syncNonTransactional = syncNonTransactional;
 
-      if (scheduledExecutor != null && syncTimeout > 0) {
-         this.syncTimer = new PageSyncTimer(this, scheduledExecutor, ioExecutor, syncTimeout);
-      } else {
-         this.syncTimer = null;
-      }
+      this.timedWriter = createPageTimedWriter(scheduledExecutor, syncTimeout);
 
       this.cursorProvider = storeFactory.newCursorProvider(this, this.storageManager, addressSettings, executor);
 
       this.usingGlobalMaxSize = pagingManager.isUsingGlobalSize();
    }
 
+   protected PageTimedWriter createPageTimedWriter(ScheduledExecutorService scheduledExecutor, long syncTimeout) {
+      if (scheduledExecutor != null && syncTimeout > 0) {
+         PageTimedWriter localWriter = new PageTimedWriter(pageSize, storageManager, this, scheduledExecutor, executor, syncNonTransactional, syncTimeout);
```

Comment:
_scheduledExecutor_ and _syncTimeout_ are passed in from the constructor, but others like _executor_ are not and instead accessed from the fields, would be nice to make it consistent if possible.

Reply:
I created this method just so I could replace it on a UnitTest.

I don't need a field created for anything else... I could create a field for these and access it  through, but I prefer the way it is now. 

let if you are strong opinionated about this and I would be easy to make it a field.

Reply:
@gemmellr I will mark this as resolved (and please unresolve it if you don't agree).. I would prefer to keep these fields as part of the parameter here. These are used in a constructor for the purpose of testability.  Accessing a field before constructor is done did not seem a good option to me. I will leave it as is.